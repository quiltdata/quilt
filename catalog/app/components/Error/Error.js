import PT from 'prop-types'
import * as React from 'react'
import { setPropTypes } from 'recompose'
import { Box, Typography } from '@material-ui/core'
import { styled } from '@material-ui/styles'

import { composeComponent } from 'utils/reactTools'
import { printObject } from 'utils/string'

import sand from './sand.jpg'

const Img = styled(Box)({
  backgroundImage: `url(${sand})`,
  backgroundRepeat: 'no-repeat',
  backgroundSize: 'cover',
})

// TODO add sign in
export default composeComponent(
  'Error',
  setPropTypes({
    headline: PT.node,
    detail: PT.node,
    object: PT.object,
  }),
  ({
    detail = 'Check network connection and login',
    headline = 'Something went wrong',
    object,
  }) => (
    <React.Fragment>
      <Typography variant="h4" gutterBottom>
        {headline}
      </Typography>
      <Typography variant="body1">{detail}</Typography>
      <Img height={600} mt={2} />
      {!!object && <Box component="pre">{printObject(object)}</Box>}
    </React.Fragment>
  ),
)
