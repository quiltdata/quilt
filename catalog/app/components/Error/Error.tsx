import * as React from 'react'
import * as M from '@material-ui/core'
import { styled } from '@material-ui/styles'

import { printObject } from 'utils/string'

import sand from './sand.jpg'

const Img = styled(M.Box)({
  backgroundImage: `url(${sand})`,
  backgroundRepeat: 'no-repeat',
  backgroundSize: 'cover',
})

interface ErrorProps {
  headline?: React.ReactNode
  detail?: React.ReactNode
  object?: {}
}

export default function Error({
  detail = 'Check network connection and login',
  headline = 'Something went wrong',
  object,
}: ErrorProps) {
  return (
    <>
      <M.Typography variant="h4" gutterBottom>
        {headline}
      </M.Typography>
      <M.Typography variant="body1">{detail}</M.Typography>
      <Img height={600} mt={2} />
      {!!object && <M.Box component="pre">{printObject(object)}</M.Box>}
    </>
  )
}
