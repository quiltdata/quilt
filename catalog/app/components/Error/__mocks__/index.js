import { mockComponent } from 'testing/util';

export default mockComponent('Error', {
  children: ['headline', 'detail'],
});
