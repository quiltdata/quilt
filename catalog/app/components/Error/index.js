/* Error */
import PT from 'prop-types';
import React from 'react';
import { Col, Row } from 'react-bootstrap';
import { setPropTypes } from 'recompose';

import ImageRow from 'components/ImageRow';
import { composeComponent } from 'utils/reactTools';
import { printObject } from 'utils/string';

import sand from './sand.jpg';

// TODO add sign in
export default composeComponent('Error',
  setPropTypes({
    headline: PT.node,
    detail: PT.node,
    object: PT.object,
  }),
  ({
    detail = 'Check network connection and login',
    headline = 'Something went wrong',
    object,
  }) => (
    <div>
      <h1>{headline}</h1>
      <h2>{detail}</h2>
      <ImageRow height="600px" src={sand} />
      {!!object && (
        <Row>
          <Col xs={12}>
            <pre>
              {printObject(object)}
            </pre>
          </Col>
        </Row>
      )}
    </div>
  ));
