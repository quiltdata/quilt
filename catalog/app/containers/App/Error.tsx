import * as React from 'react'
import { useRouteError, isRouteErrorResponse } from 'react-router-dom'

import Error from 'components/Error'

export default function ErrorElement() {
  let error = useRouteError()
  const message = isRouteErrorResponse(error)
    ? `${error.status} ${error.statusText}`
    : (error as Error).message || error
  return <Error detail={message} />
}
