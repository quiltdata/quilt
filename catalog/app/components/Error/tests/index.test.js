import React from 'react';
import { mount } from 'enzyme';

import Error from '..';

jest.mock('react-bootstrap');

jest.mock('components/ImageRow');
jest.mock('constants/config');


describe('components/Error', () => {
  it('should render properly with default values', () => {
    const html = mount(<Error />).render();
    expect(html.find('h1').text()).toMatch('Something went wrong');
    expect(html.find('h2').text()).toMatch('Check network connection and login');
    expect(html.find('pre')).toHaveLength(0);
    expect(html).toMatchSnapshot();
  });

  it('should render properly with non-default values', () => {
    const obj = { a: { b: 2 } };
    const html = mount(
      <Error
        headline="Headline"
        detail="test"
        object={obj}
      />
    ).render();
    expect(html.find('h1').text()).toMatch('Headline');
    expect(html.find('h2').text()).toMatch('test');
    expect(html.find('pre').text()).toMatch('{\n  "a": {\n    "b": 2\n  }\n}');
    expect(html).toMatchSnapshot();
  });
});
