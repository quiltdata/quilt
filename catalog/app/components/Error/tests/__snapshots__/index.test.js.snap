// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Error should render properly with default values 1`] = `
<div>
  <h1>
    Something went wrong
  </h1>
  <h2>
    Check network connection and login
  </h2>
  <component-image-row
    height="600px"
    src="IMAGE_MOCK"
  />
</div>
`;

exports[`components/Error should render properly with non-default values 1`] = `
<div>
  <h1>
    Headline
  </h1>
  <h2>
    test
  </h2>
  <component-image-row
    height="600px"
    src="IMAGE_MOCK"
  />
  <component-row>
    <component-col
      xs="12"
    >
      <pre>
        {
  "a": {
    "b": 2
  }
}
      </pre>
    </component-col>
  </component-row>
</div>
`;
